tokio-trace-subscriber/src/registry.rs

```diff
@@ -44,6 +47,13 @@ pub trait RegisterSpan {
         value: &dyn IntoValue,
     ) -> Result<(), AddValueError>;
 
+    /// Indicates that `span` follows from `follows.
+    fn add_follows_from(&self, span: &Id, follows: Id);
+
+    /// Queries the registry for an iterator over the IDs of the spans that
+    /// `span` follows from.
+    fn get_follows_from(&self, span: &Id) -> Self::FollowsFrom;
```

Comment:
Are there other ways of querying these relationships that ought to be part of the registry API?

Reply:
Should the subscriber interface also have a similar function? I kept it out because I didn't want to add too many more methods to that trait...

Reply:
> Are there other ways of querying these relationships that ought to be part of the registry API?

I'm thinking the `follows_from` family of methods should be named `succeeds`/`proceeds` to more clearly delineate ordering. Alternatively—and this might be a infeasible—but a `PartialOrd` implementation could be helpful.

> 

Should the subscriber interface also have a similar function? I kept it out because I didn't want to add too many more methods to that trait.

No, probably not, but I can be convinced otherwise.

Reply:
@davidbarsky 
> I'm thinking the `follows_from` family of methods should be named `succeeds`/`proceeds `to more clearly delineate ordering. 

That's reasonable --- this naming is per https://github.com/tokio-rs/tokio/issues/561#issue-353150714, but I'm open to renaming it.

> Alternatively—and this might be a infeasible—but a `PartialOrd` implementation could be helpful.

Having something like this would certainly be ergonomic, but I don't think it's immediately possible in the current scheme --- it's the _subscriber_ that knows these relationships, not the spans themselves.

Reply:
> That's reasonable --- this naming is per tokio-rs/tokio#561 (comment), but I'm open to renaming it. I'm not sure how I feel about succeeds here, though, since that seems easy to confuse with a notion of success/failure...

True. What about “prior/subsequent?”

> Having something like this would certainly be ergonomic, but I don't think it's immediately possible in the current scheme --- it's the subscriber that knows these relationships, not the spans themselves. That said, I'll keep thinking about the potential of doing that.

Good point, didn't consider that that part.

Reply:
> > That's reasonable --- this naming is per tokio-rs/tokio#561 (comment), but I'm open to renaming it. I'm not sure how I feel about succeeds here, though, since that seems easy to confuse with a notion of success/failure...
>
> True. What about “prior/subsequent?”

That works for me!

> > [...] I don't think it's immediately possible in the current scheme --- it's the subscriber that knows these relationships, not the spans themselves. 
>
> Good point, didn't consider that that part.

I am considering changing the `SpanRef` type in the `tokio-trace-subscriber` trait to have a reference back to the registry for querying it to get span data, so it may be possible to implement something like that for `SpanRef`s in the more batteries-included crate...but that'll be its own branch.